source/routing/get_stop_times.cpp

```diff
@@ -60,9 +60,9 @@ std::vector<datetime_stop_time> get_stop_times(const routing::StopEvent stop_eve
         auto st = next_st.next_stop_time(stop_event, jpp_idx,
                                          dt, clockwise, rt_level,
                                          accessibilite_params.vehicle_properties, true);
-
         if (st.first) {
-            next_requested_dt.push({jpp_idx, st.first, st.second});
+            auto diff = clockwise ? st.first->get_boarding_duration() : -1 * st.first->get_alighting_duration();
+            next_requested_dt.push({jpp_idx, st.first, st.second + diff});
```

Comment:
This part of my code was vaguely stupid... I added some unit tests and I'm going to add more. And while stop_schedules and route_schedules are sorting their departures passages wasn't. Thanks for the heads up on that one :).